extensions/compositions/compositions.preview.runtime.ts

```diff
@@ -10,20 +11,31 @@ export class CompositionsPreview {
     private preview: PreviewPreview
   ) {}
 
-  render(componentId: string, modules: any) {
+  render(componentId: string, modules: PreviewModule) {
     if (!modules.componentMap[componentId]) return;
-    const composition = this.getActiveComposition(modules.componentMap[componentId][0]);
-    modules.mainModule.default(composition);
+
+    const compositions = this.selectPreviewModel(componentId, modules);
+    const active = this.getActiveComposition(compositions);
+
+    modules.mainModule.default(active);
+  }
+
+  /** gets relevant information for this preview to render */
+  selectPreviewModel(componentId: string, previewModule: PreviewModule) {
```

Comment:
merges composition files into one object. (within the same component)
as per Ran, compositions will have unique names even between different files